src/ast/passes/parser.h

```diff
@@ -14,20 +14,21 @@ namespace bpftrace::ast {
 // AllParsePasses returns a vector of passes representing all parser passes, in
 // the expected order. This should be used unless there's a reason not to.
 inline std::vector<Pass> AllParsePasses(
-    std::vector<std::string> &&extra_flags = {})
+    std::vector<std::string> &&extra_flags = {},
+    bool listing = false)
```

Comment:
I'm trying to remove "listing" as a parameter to the passes in general, as it's really a "please do a completely different thing" flag, rather than one that controls the behavior more narrowly.

I think you can just drop this from `AllParsePasses` and keep it, since the listing behavior is a manual construction in `main` anyways.